tls/_common/prf.py

```diff
@@ -0,0 +1,69 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.hmac import HMAC
+
+
+def _p_hash(hash_algorithm, secret, seed, output_length):
+    """
+    A seed expansion function that uses a single hash function to expand a
+    secret and seed into the number of bytes specified by output_length.
+    """
+    result = b""
+    i = 1
+    while len(result) < output_length:
+        h = HMAC(secret, hash_algorithm, default_backend())
+        h.update(_a(secret, hash_algorithm, i, seed))
+        h.update(seed)
+        result += h.finalize()
```

Comment:
I ran `print(timeit.timeit("TestPRFVectors.test_sha256_100bytes", setup="from tls._common.test.test_prf import TestPRFVectors"))` with the following versions of `_p_hash()`:

```
result = bytearray()
i = 1
while len(result) < output_length:
    h = HMAC(secret, hash_algorithm, default_backend())
    h.update(_a(secret, hash_algorithm, i, seed))
    h.update(seed)
    result.extend(h.finalize())
    i += 1
return bytes(result[:output_length])
```
gives **0.06398137410483333** as the mean of 3 runs.


```    
    result = b""
    i = 1
    while len(result) < output_length:
        h = HMAC(secret, hash_algorithm, default_backend())
        h.update(_a(secret, hash_algorithm, i, seed))
        h.update(seed)
        result += h.finalize()
        i += 1
    return result[:output_length]
``` 
gave **0.0521132946014** as the mean of 3 runs.


```
    result = []
    i = 1
    result_length = 0
    while result_length < output_length:
        h = HMAC(secret, hash_algorithm, default_backend())
        h.update(_a(secret, hash_algorithm, i, seed))
        h.update(seed)
        temp_result = h.finalize()
        result.append(temp_result)
        result_length += len(temp_result)
        i += 1
    result = b''.join(result)
    return result[:output_length]
``` 
gave **0.07136265436806667** as mean of 3 runs.